src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.CheckOrReachability.cs

```diff
@@ -0,0 +1,956 @@
+                ImmutableArray<BoundPropertySubpattern> propertySubpatterns = recursivePattern.Properties;
+                if (!propertySubpatterns.IsDefault)
+                {
+                    for (int i = 0; i < propertySubpatterns.Length; i++)
+                    {
+                        BoundPattern pattern = propertySubpatterns[i].Pattern;
+                        using SetsOfOrCases setOfOrCases = RewriteToSetsOfOrCases(pattern);
+                        if (!setOfOrCases.IsDefault)
+                        {
+                            foreach (OrCases orSet in setOfOrCases.Set)
+                            {
+                                OrCases resultOrSet = result.StartNewOrCases();
+                                foreach ((BoundPattern resultPattern, SyntaxNode? syntax) in orSet.Cases)
+                                {
+                                    ImmutableArray<BoundPropertySubpattern> properties = recursivePattern.Properties;
+                                    BoundRecursivePattern resultRecursivePattern = recursivePattern
+                                        .WithProperties(properties.SetItem(i, properties[i].WithPattern(resultPattern)));
+
+                                    resultOrSet.Add(resultRecursivePattern, syntax);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // If any of the nested positional patterns can be expanded, we carry those on.
+                // For example, in `(A, ... )`, when we found multiple `or` patterns in `A` to be expanded
+                // For each such nested expansion, we'll produce an `(<expansion>, ... )` expansion
+                ImmutableArray<BoundPositionalSubpattern> positionalSubpatterns = recursivePattern.Deconstruction;
+                if (!positionalSubpatterns.IsDefault)
+                {
+                    for (int i = 0; i < positionalSubpatterns.Length; i++)
+                    {
+                        BoundPattern pattern = positionalSubpatterns[i].Pattern;
+                        using SetsOfOrCases setOfOrCases = RewriteToSetsOfOrCases(pattern);
+                        if (!setOfOrCases.IsDefault)
+                        {
+                            foreach (OrCases orSet in setOfOrCases.Set)
+                            {
+                                OrCases resultOrSet = result.StartNewOrCases();
+                                foreach ((BoundPattern resultPattern, SyntaxNode? syntax) in orSet.Cases)
+                                {
+                                    ImmutableArray<BoundPositionalSubpattern> deconstruction = recursivePattern.Deconstruction;
+                                    BoundRecursivePattern resultRecursivePattern = recursivePattern
+                                        .WithDeconstruction(deconstruction.SetItem(i, deconstruction[i].WithPattern(resultPattern)));
```

Comment:
>.WithDeconstruction(deconstruction.SetItem(i, deconstruction[i].WithPattern(resultPattern))); [](http://example.com/codeflow?start=40&length=93)

Similar comment here, normalization would give us a guarantee that there is only one meaningful positional sub-pattern to look at.